climpred/relative_entropy.py

```diff
@@ -0,0 +1,343 @@
+import warnings
+from collections import OrderedDict
+
+import matplotlib.pyplot as plt
+import numpy as np
+import xarray as xr
+from eofs.xarray import Eof
+
+
+def _relative_entropy_formula(sigma_b, sigma_x, mu_x, mu_b, neofs):
+    """
+    Computes the relative entropy formula given in Branstator and Teng, (2010).
+
+    References:
+        - Branstator, Grant, and Haiyan Teng. “Two Limits of Initial-Value
+            Decadal Predictability in a CGCM.” Journal of Climate 23, no. 23
+            (August 27, 2010): 6292–6311. https://doi.org/10/bwq92h.
+        - Kleeman, Richard. “Measuring Dynamical Prediction Utility Using
+            Relative Entropy.” Journal of the Atmospheric Sciences 59, no. 13
+            (July 1, 2002): 2057–72. https://doi.org/10/fqwxpk.
+
+    Args:
+        sigma_b (xr.DataArray): covariance matrix of baseline distribution
+        sigma_x (xr.DataArray): covariance matrix of forecast distribution
+        mu_b (xr.DataArray): mean state vector of the baseline distribution
+        mu_x (xr.DataArray): mean state vector of the forecast distribution
+        neofs (int): number of EOFs used
+
+    Returns:
+        R (float): relative entropy
+        dispersion (float): dispersion component
+        signal (float): signal component
+    """
+    fac = 0.5
+    dispersion = fac * (np.log(np.linalg.det(sigma_b) / np.linalg.det(sigma_x))
+                        + np.trace(sigma_x / sigma_b) - neofs)
+
+    # https://stackoverflow.com/questions/7160162/left-matrix-division-and-numpy-solve
+    # (A.T\B)*A
+    x, resid, rank, s = np.linalg.lstsq(
+        sigma_b, mu_x - mu_b)  # sigma_b \ (mu_x - mu_b)
+    signal = fac * np.matmul((mu_x.values - mu_b.values), x)
+    R = dispersion + signal
+    return R, dispersion, signal
+
+
+def _bootstrap_dim(control, lead_years, time_dim='initialization', dim='member', dim_label=list(np.arange(10))):
+    """
+    Add a `len(dim_label)` dimension `dim` to uninitialized control with time_dim by bootstrapping.
+
+    """
+    c_start = 0
+    c_end = control[time_dim].size
+    time = np.arange(1, 1 + lead_years)
+
+    def isel_years(control, year_s, m=None, length=lead_years):
+        new = control.isel({time_dim: slice(year_s, year_s + length - 0)})
+        new = new.rename({time_dim: 'time'})
+        if isinstance(new, xr.DataArray):
+            new['time'] = time
+        elif isinstance(new, xr.Dataset):
```

Comment:
I don't think these checks are necessary from what I've seen. You should just be able to do:

```python
new['time'] = time
```
without the check on DataArray vs. Dataset.